lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2470,6 +2470,7 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
   void visitBoundVariable(VarDecl *VD) {
     TC.validateDecl(VD);
+    TC.addImplicitDynamicAttribute(VD);
```

Comment:
Note that typeCheckDecl() only gets called for stuff in the primary file(s). Is addImplicitDynamicAttribute() only for debugging? Even still, does it matter if its not called on things in other files that are referenced from the primary file?

Reply:
:/

The right way do this would be to check in the isDynamic request if the decl is from the "current module" and add it if requested.